bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -351,9 +355,31 @@ void ReplicaImp::removeDuplicatedRequestsFromRequestsQueue() {
   while (first != nullptr && !clientsManager->canBecomePending(first->clientProxyId(), first->requestSeqNum())) {
```

Comment:
Maybe I'm missing something, but still, this loop will stop at the first request that can become pending.

Reply:
I did not change this loop, because I saw that we check the condition prior to insertion [ReplicaImp.cpp:261](https://github.com/vmware/concord-bft/blob/ff6416fa3d0f31c961a8d377460da61e1c466867/bftengine/src/bftengine/ReplicaImp.cpp#L261)